theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -307,7 +308,7 @@ def _optimize_impl(
                 f"Error: {info.last_err.mean().item()}"
             )
 
-        if backward_mode == BackwardMode.FULL:
+        if backward_mode in [BackwardMode.FULL, BackwardMode.DLM]:
```

Comment:
Should DLM be handled separately where you have `with torch.no_grad()`?

Reply:
The forward and backward functions inside TheseusLayerDLMForward (where this code will be called) automatically have a torch.no_grad(), so there's no need to explicitly change the behavior here.